### PR TITLE
Extract `AccountFollowable` concern for `Account.followable_by` scope

### PR DESCRIPTION
**Pull request by mjankowski**
Pulled out from https://github.com/mastodon/mastodon/pull/26093

This scope was sort of complicated and hard to follow and parse visually.

This change pulls the whole thing out into a concern which does nothing but build the scope back up from better-named private methods. From my local testing, a) the generated sql hasn't changed, b) the specs which exercise this method seem fine with the change.

### CODE REVIEW

File: app/models/concerns/account_followable.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module AccountFollowable
```
Comment (mjankowski) [inline app/models/concerns/account_followable.rb:3]: Instead of this being a concern, could the whole thing be a new class?

Something like `FollowableFilter.new(account).accounts` as the scope in the class?

Comment (ClearlyClaire) [reply app/models/concerns/account_followable.rb:3]: It is also only ever used by `AccountSuggestions::GlobalSource#scope` so maybe it could be moved to this class.

Comment (mjankowski) [reply app/models/concerns/account_followable.rb:3]: It's also in `AccountSuggestions::SettingSource`, but they both inherit from same base class and use it in similar ways ... will look at moving in there as well.

**Comment by mjankowski** [comment]
Updated this to move the whole query generation to a new class, and remove the scope from `Account` since its only ever used in the suggestion area. I think the goal of making it less dense and more readable is better with this approach than just the extraction to the concern.

The suggestions classes, and maybe the new query class could probably be further simplified. I suspect that at least part of the query generation does not need the full low level arel treatment it's currently using, but haven't verified in detail.

**Comment by mjankowski** [comment]
Rebased this again after the introduction of spec coverage in https://github.com/mastodon/mastodon/pull/28689

Current version of this is pull out the scope creation to a concern module, building two smaller scopes with the conditions, and then combining them. I think this is a net-improvement as-is, but would like feedback. If this is not good to go, the other idea I have is to move (and expand) the coverage to be directly on `AccountSuggestions::Source` (the only place this is used), and then shift the scope generation into that class and out of `Account`.

**Comment by ClearlyClaire** [comment]
Taking a step back from this query, it heavily uses `arel` to build queries of the form `LEFT OUTER JOIN … ON … WHERE id = NULL` , which can be expressed in a somewhat simpler (and sometimes more efficient) way with `WHERE NOT EXISTS (SELECT 1 FROM … WHERE …)`. Unfortunately, neither can be easily written with ActiveRecord itself and would require either raw SQL or extensive use of `arel`.

Given that this query is only ever used in `AccountSuggestions::Source`, I would move it there and use raw SQL, as this file already does for another part of the query, and in a similar way `AccountSuggestions::FriendsOfFriendsSource` already does.

Something like:
```ruby
  def followable_by(account)
    Account.where.not('EXISTS (SELECT 1 FROM follows f WHERE f.target_account_id = accounts.id AND f.account_id = :id)', id: account.id)
           .where.not('EXISTS (SELECT 1 FROM follow_requests f WHERE f.target_account_id = accounts.id AND f.account_id = :id)', id: account.id)
  end

**Comment by mjankowski** [comment]
Great. I'll move the coversage over to account suggesstion spec, expand it a bit, and then explore something along those lines.

**Comment by mjankowski** [comment]
Closing this, goal accomplished by linked PR.
